### PR TITLE
Language settings & search parameter support for Java client

### LINKED ISSUE #770: [v1.10.0] Language settings & search parameter
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/302)*

### Explanation of the feature

Changes
- new settings `localizedAttributes` and the sub-methods associated to it
- new search parameter `locales`

Usage: https://meilisearch.notion.site/v1-10-Language-settings-usage-26c5d98b553349d9abacbe7aff698e4e

### TODO

- [ ] Add a new search parameter named `locales` to the `search` methods (GET and POST)
- [ ] Ensure the `updateSettings` route can accept the new `localizedAttributes` setting
- [ ] Create new methods `getLocalizedAttributes`, `updateLocalizedAttributes` and `resetLocalizedAttributes` corresponding to the sub settings `GET/PUT/DELETE` routes
- [ ] Add tests
- [ ] Add code samples for the feature:
    ```js
    search_parameter_reference_locales_1: |-
        client.index('INDEX_NAME').search('進撃の巨人', { locales: ['jpn'] })
    get_localized_attribute_settings_1: |-
        client.index('INDEX_NAME').getLocalizedAttributes()
    update_localized_attribute_settings_1: |-
        client.index('INDEX_NAME').updateLocalizedAttributes([
            { attributePatterns: ['jpn'], locales: ['*_ja'] },
        ])
    reset_localized_attribute_settings_1: |-
        client.index('INDEX_NAME').resetLocalizedAttributes()
    ```

### PR DESCRIPTION
**Pull request by Barabasbalazs**
# Pull Request

## Related issue
Fixes #770

## What does this PR do?
- Implements support for the _localizedAttributes_ setting, _locales_ search parameter and updates the tests accordingly.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by curquiza** [comment]
@Barabasbalazs thanks for your PR

Can you fix the tests and the linter before I review?

**Comment by Barabasbalazs** [comment]
@curquiza fixed the linting issues

**Comment by curquiza** [comment]
@Barabasbalazs tests look good ✅ 

Can you fix the git conflicts, since your other PR has been merged?

**Comment by Barabasbalazs** [comment]
@curquiza should be good now

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -837,3 +837,18 @@ distinct_attribute_guide_filterable_1: |-
 distinct_attribute_guide_distinct_parameter_1: |-
   SearchRequest searchRequest = SearchRequest.builder().q("white shirt").distinct("sku").build();
   client.index("products").search(searchRequest);
+search_parameter_reference_locales_1: |-
+  SearchRequest searchRequest = SearchRequest.builder().q("進撃の巨人").locales(new String[]{"jpn"}).build();
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:841]:
```suggestion
  SearchRequest searchRequest = SearchRequest.builder().q("QUERY TEXT IN JAPANESE").locales(new String[]{"jpn"}).build();
```

**Review by curquiza — APPROVED** [review]
bors merge

**Pull request merged**